pyprusalink/types.py

```diff
@@ -0,0 +1,131 @@
+from typing import TypedDict, Optional
+from enum import Enum
+
+"""Types of the v1 API. Source: https://github.com/prusa3d/Prusa-Link-Web/blob/master/spec/openapi.yaml"""
+
+
+class PrusaLinkError(Exception):
+    """Base class for PrusaLink errors."""
+
+
+class InvalidAuth(PrusaLinkError):
+    """Error to indicate there is invalid auth."""
+
+
+class Conflict(PrusaLinkError):
+    """Error to indicate the command hit a conflict."""
+
+
+class Capabilities(TypedDict):
+    """API Capabilities"""
+
+    upload_by_put: Optional[bool]
```

Comment:
Since we are supporting Python 3.11+ let's use union type instead (`bool | None`).

Reply:
Switched to union types: https://github.com/home-assistant-libs/pyprusalink/pull/63/commits/854007493a5ef92f4b31954dca878569059eebea